Gems/GradientSignal/Code/Source/Editor/EditorGradientBakerComponent.cpp

```diff
@@ -117,6 +117,10 @@ namespace GradientSignal
         const AZ::Vector3 pixelToBoundsScale(
             inputBoundsExtentsX / static_cast<float>(imageResolutionX), inputBoundsExtentsY / static_cast<float>(imageResolutionY), 0.0f);
 
+        // Generate a set of input positions that are inside the bounds along with
+        // their corresponding x,y indices
+        AZStd::vector<AZ::Vector3> inputPositions;
```

Comment:
+1 since this is meant as an optimization, reserving space in containers will help reduce reallocations and improve performance, especially because it's pushing one element at a time and the data set could be large.

Reply:
Actually, it should be `.reserve(imageResolutionX * imageResolutionY)`, not constructed with a starting size. Otherwise you'll initialize it with valid values, which isn't what you want here, since you're filling it with specific values, and not necessarily all the way to capacity.

Reply:
Will do!